src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MemoryPoolViewBufferScope.cs

```diff
@@ -36,8 +36,13 @@ public MemoryPoolViewBufferScope(ArrayPool<ViewBufferValue> viewBufferPool, Arra
         }
 
         /// <inheritdoc />
-        public ViewBufferValue[] GetSegment()
+        public ViewBufferValue[] GetPage(int size)
         {
+            if (size <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(size));
```

Comment:
This is a `public` class in an `Internal` namespace. I don't get why anything but `Debug` assertions are needed.